### DISCUSSION TITLE
Segmentation questions

### DISCUSSION THREAD

**Discussion by lauren-herbine** [Riverscapes/riverscapes-tools#326] [VBET] [answered]
## General Question
As I am thinking about RS metrics and how they're calculated... I keep on running into the question of riverscape segmentation. 
**How will we be segmenting reaches, valley bottoms, centerlines, sausage links, etc. for each metric?** 
**Can some metrics be calculated at the smallest reach length (ie one vb sausage or the like) while others are calculated for multiple segments?** 

Is [this](https://github.com/Riverscapes/riverscapes-tools/issues/225#issuecomment-843279878) method of calculating metrics at multiple scales achievable?
![image](https://user-images.githubusercontent.com/73319684/124160528-944dc000-da59-11eb-9185-12816a4ca02a.png)
Notebaert B, Piegay H. 2013. Multi-scale factors controlling the pattern of floodplain width at a network scale: The case of the Rhone basin, France. Geomorphology 200 : 155–171. DOI: 10.1016/j.geomorph.2013.03.014

## Example where this would be useful
_Planform sinuosity calculations_

![segmentation with sinuosity](https://user-images.githubusercontent.com/73319684/124163929-5e123f80-da5d-11eb-9336-decb57f4a9e4.png)

Here, when looking at the riverscape at a much smaller scale (zoomed out), it appears as though the sinuosity would be sinuous/meandering (and a quick Google Earth calculation gives us a value of 1.37 for a ~4,500 m vb segment (white circle in sketch below)).  If segmented at the road crossing, however (light yellow circle in sketch below and red borders in sketch above), we get a planform sinuosity value of 1.13, which would indicate low sinuousity. 

![scaling segmentation](https://user-images.githubusercontent.com/73319684/124164732-425b6900-da5e-11eb-864a-e08bf6220d9d.png)

This would not be an issue if each of these smallest vb segments (i.e. ones similar in size to the segment created by the road crossings) were each different river styles. In this case, however, it appears to me that the river style on the mainstem would actually stay the same until confinement increases moving up the catchment.

I'm attempting to highlight that calculating sinuosity at smaller scales ("zoomed out") than other metrics that are calculated at our most segmented scale (i.e. when there are many road crossings) would be helpful in many cases.

## Currently...

_Some discussions where this has been touched on:_
#225 
#236 

As I understand it, there are currently four main ways we are/could be segmenting the riverscapes:
1. NHD+HR channel segmentation (default) 
2. additional segmentation (300m, land ownership, roads and rail)
3. VBET CL segmentation
4. At ecoregions (#236)

Is this correct?
Any insights @KellyMWhitehead ?

## TL,DR

Basically, how hard is it to calculate at different scales? Can we do calculations on the fly for different scales? Are we, as a group, leaning toward any specific segmentation process?

**Comment by lauren-herbine** [comment]
Thoughts @joewheaton, @philipbaileynar, @shelbysawyer ?

**Comment by joewheaton** [comment]
@KellyMWhitehead has implemented a version of segmentation and we have to decide what we want segmentation choices to be.

![image](https://user-images.githubusercontent.com/9723905/128252088-110a14d1-04f7-4702-8c54-54a1e05baeb1.png)

![image](https://user-images.githubusercontent.com/9723905/128252678-d90a3200-cbf0-44d1-aba9-59345d45c3f8.png)


We

**Comment by joewheaton** [comment]
If we think about DGOs... they are node centered calculations where the length scale over which makes sense (e.g. calculate sinuosity based no on a channel length within the segmented, but by moving upstream and downstream at a fixed length).
![image](https://user-images.githubusercontent.com/9723905/128253581-72069ab6-b214-419e-889e-34afefab1c67.png)

**Reply by lauren-herbine** [reply]
This could be good, especially if we are going to lean toward finer segmentation (such as at every fan controlled reach).

**Comment by lauren-herbine** [comment]
I'm a little concerned about broader valleys with a lot of connectors/intermittents/artificial paths/etc. This would be a lot of ag-y areas.

![image](https://user-images.githubusercontent.com/73319684/128391427-a21b18f1-f531-45cf-9540-2d114c552c76.png)
An example from the [St. Vrain](https://data.riverscapes.xyz/#/PUBLIC_DATA/bf8bf41f-c8e0-432c-80c7-d22bc48138d9) project that just came out. Would we necessarily want all of these little pockets of plausible valley bottom extent creating many, many small segments?

Some of these segments are 10-60 meters long:
![image](https://user-images.githubusercontent.com/73319684/128391989-c40fb50f-cd7b-4975-9c5f-7fd9c631a605.png)

Perhaps I am interpreting this incorrectly... but just a concern I would have

**Reply by joewheaton** [reply]
Good point @lauren-herbine. This is why the segmentation choices, and segmentation sanitization (e.g. potential minimum sizes) are going to be important to think about. Keep flagging the types of issues here and thinking about what you really want to be reach typing.

**Comment by joewheaton** [comment, marked as answer]
This issue is now consolidated over in #414
